test/puppetlabs/trapperkeeper/optional_deps_test.clj

```diff
@@ -32,6 +32,83 @@
                                             couplet))))
 
 (deftest optional-deps-test
+  (testing "when using defservice"
+    (testing "with a protocol"
+      (defservice poetry-svc-w-protocol
+        PoetryService
+        {:required [HaikuService]
+         :optional [SonnetService]}
+        (init [this ctx] ctx)
+        (get-haiku [this] "haiku")
+        (get-sonnet [this] "sonnet"))
```

Comment:
I've generally tried to avoid putting defs in tests, but I don't feel strongly about it. If it seems likely we'll have other tests also using the same service it's probably worth lifting it out.